### PR TITLE
Fix(eos_designs): Failure with OSPF underlay and mlag_peer_l3_vlan == mlag_peer_vlan

### LINKED ISSUE #1841: Bug(eos_designs): playbook fails if mlag_peer_l3_vlan == mlag_peer_vlan
### Issue Summary

playbook fails with 

```
fatal: [DUT -> localhost]: FAILED! => 
  msg: 'dict object' has no attribute 'mlag_peer_l3_vlan'
```
when
```
 when mlag_peer_l3_vlan in [None, False, mlag_peer_vlan]
```

The issue is that in router_ospf.j2, we check the parent attribute instead of the vlan:
```jinja
{% if switch.mlag_l3 is arista.avd.defined(true) %}
        - Vlan{{ switch.mlag_peer_l3_vlan }}
{% endif %}
```

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
Configure an mlag leaf in molecule with:

mlag_peer_l3_vlan: mlag_peer_vlan
```

### Relevant log output

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by pvinci-arista**
## Change Summary
molecule test to demonstrate failure

## Related Issue(s)

Fixes #1841 

## Component(s) name

`arista.avd.eos_designs`

## How to test

ANSIBLE_OPTIONS='-vvv' MOLECULE=eos_designs_unit_tests make run

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [X] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/templates/underlay/ospf/router-ospf.j2 [hunk 1]
```diff
@@ -10,7 +10,7 @@ router_ospf:
         - {{ link.interface }}
 {%     endif %}
 {% endfor %}
-{% if switch.mlag_l3 is arista.avd.defined(true) %}
+{% if switch.mlag_peer_l3_vlan is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/underlay/ospf/router-ospf.j2]: This is looking for `true` in the vlan variable, which would never work :)

```suggestion
{% if switch.mlag_peer_l3_vlan is arista.avd.defined %}
```

**Comment by tgodaA** [comment]
Please run the molecule once again for **eos_designs_unit_tests**.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/MLAG_OSPF_TESTS.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    uplink_switches: ["DC1-SPINE1"]
+    uplink_interfaces: ["Ethernet1"]
+  node_groups:
+    MLAG_OSPF_L3LEAF1:
+      nodes:
+        MLAG-OSPF-L3LEAF1A:
+          id: 4
+          bgp_as: 65161
+          mgmt_ip: 192.168.201.114/24
+          uplink_switch_interfaces: [Ethernet16]
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/MLAG_OSPF_TESTS.yml]: `Ethernet16` has been taken on Spine1 for some recently merged PR:
https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_FABRIC.yml#L228

(and so has `Ethernet17`) - this causes one of the molecule tests to fail in CI

Could you use `Ethernet18`and push back the PR?

**Review by gmuloc — CHANGES_REQUESTED** [review]
Can you also add the test in the `eos_designs_unit_tests_v4.0` once you have fixed the conflicting Ethernet interface?

**Comment by carlbuchmann** [comment]
> Can you also add the test in the `eos_designs_unit_tests_v4.0` once you have fixed the conflicting Ethernet interface?

@gmuloc @pvinci-arista - note, that we don't need to update the `eos_designs_unit_tests_v4.0` scenario at this time - we will merge them back into one scenario once we are done refactoring.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/MLAG_OSPF_TESTS.yml (hunk 2 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/MLAG_OSPF_TESTS.yml]: thanks @gmuloc - resolved in [9be9059](https://github.com/aristanetworks/ansible-avd/pull/1842/commits/9be90596c9ed5d89aff545bd28d461c0e01c1c31)

**Comment by gmuloc** [comment]
> > Can you also add the test in the `eos_designs_unit_tests_v4.0` once you have fixed the conflicting Ethernet interface?
> 
> @gmuloc @pvinci-arista - note, that we don't need to update the `eos_designs_unit_tests_v4.0` scenario at this time - we will merge them back into one scenario once we are done refactoring.

Noted! apologies for the comment

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**